packages/i18n-utils/src/test/localize-url.js

```diff
@@ -55,6 +55,34 @@ describe( '#localizeUrl', () => {
 		} );
 	} );
 
+	test( 'should not accept host-only matches if skipHostLookup is true', () => {
+		expect( localizeUrl( 'https://wordpress.com/some/path', 'es', false, false ) ).toEqual(
+			'https://wordpress.com/es/some/path/'
+		);
+		expect( localizeUrl( 'https://wordpress.com/some/path', 'es', false, true ) ).toEqual(
+			'https://wordpress.com/some/path'
+		);
+		expect( localizeUrl( 'https://wordpress.com/some/path', 'es', true, true ) ).toEqual(
+			'https://wordpress.com/some/path'
+		);
+		expect( localizeUrl( 'https://wordpress.com/some/path', 'es', true, false ) ).toEqual(
+			'https://wordpress.com/es/some/path/'
+		);
```

Comment:
Thanks for the tests.. this is odd:

```suggestion
		expect( localizeUrl( 'https://wordpress.com/some/path', 'es' ) ).toEqual(
			'https://wordpress.com/some/path/'
		);
```

This test shouldn't be failing but it is even without your change, I thought this function was safe to run on variables containing unknown urls but that doesn't look to be the case for WordPress.com urls.

The localizeUrl function seems to exhibit the behavior you need for WordPress.com urls already. Maybe you don't need skipHostLookup after all?

Reply:
@yuliyan is this a bug in `localizeUrl` or do all WordPress.com paths support localizing now? Feel free to correct my assumptions here, my memory could be way off base. :shrug:

Reply:
> The localizeUrl function seems to exhibit the behavior you need for WordPress.com urls already. Maybe you don't need skipHostLookup after all?

I don't think we want the same behavior for localizePath. 
For example, without `skipHostLookup` running localizePath('/some/path') would localize it even though the path is not whitelisted and even if the user is logged in.

Reply:
> I don't think we want the same behavior for localizePath.

Definitely want the same behavior. I should be able to take any wordpress.com url in calypso and wrap it in localizeUrl without having to check if it's working right.

Reply:
@lsl, IIRC, we localize all paths on WordPress.com since Landing pages have been moved to sub directories.

The equivalent function for localizing URLs, that we have on the back-end, would also prefix `https://wordpress.com/some/path` with the locale slug, i.e. `https://wordpress.com/es/some/path`.

Reply:
Would that mean using localizeUrl inside localizePath is a mistake? It sounds like localizeUrl doesn't cover calypso routes?

Reply:
> It sounds like localizeUrl doesn't cover calypso routes?
Currently, it has URL localization mapping for `/theme`, `/themes`, `log-in`. After a quick audit, I found it lacks of mapping for `/start` and `/accept-invite` (?) routes. These should be all current Calypso routes that accept locale parameter.

As for whether it's a mistake or not - to me it seems a bit odd that `localizePath` would localize non-Calypso paths (e.g. `/support`, `/forums`, etc.), as it's currently one of the possible code paths. However, I can't really tell if that's an actual concern, given that `localizePath` should never be used for non-Calypso paths in the first place.

Reply:
> These should be all current Calypso routes that accept locale parameter.

Thanks!

> As for whether it's a mistake or not - to me it seems a bit odd that localizePath would localize non-Calypso paths (e.g. /support, /forums, etc.), as it's currently one of the possible code paths. However, I can't really tell if that's an actual concern, given that localizePath should never be used for non-Calypso paths in the first place.

Yea this is pretty hairy. Not sure what the best approach is.